### PR TITLE
add dockerfile and instruction

### PR DESCRIPTION
**Pull request by ChloeWu1**
Dockerfile and docker instruction for ov2.0 have been added. Need merge to the branch of ros2.

### CODE REVIEW

File: docker/docker_instructions_ov2.0.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+vi ~/ros2_openvino_toolkit/docker/Dockerfile
+docker build --build-arg ROS_VERSION=galactic-desktop --build-arg VERSION=galactic -t ros2_galactic_openvino_202201 .
+```
+* Build image for ros_foxy
+```
+cd ~/ros2_openvino_toolkit/docker/Dockerfile
+vi ~/ros2_openvino_toolkit/docker/Dockerfile
+docker build --build-arg ROS_VERSION=foxy-desktop --build-arg VERSION=foxy -t ros2_foxy_openvino_202201 .
+```
+**Note**:Proxy may need to set, please check dockerfile before building.
```
Comment (LewisLiuPub) [inline docker/docker_instructions_ov2.0.md:28]: Can we add a new arg for Proxy setting to command "docer build xxxx"?

File: docker/docker_instructions_ov2.0.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+* See all available models
+OMZ tools are provided for downloading and converting OMZ models in ov2022.([guide](https://pypi.org/project/openvino-dev/))
+
+```
+omz_downloader --print_all
+```
+
+* Download the optimized Intermediate Representation (IR) of model (execute once), for example:
```
Comment (LewisLiuPub) [inline docker/docker_instructions_ov2.0.md:73]: It's time to make a model.lst config file, in order to simply the model opti/installation steps?

File: docker/docker_instructions_ov2.0.md (hunk 1 above)
Comment (ChloeWu1) [reply docker/docker_instructions_ov2.0.md]: Build arg of proxy has been added. Thanks.

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+WORKDIR /root
+RUN mkdir -p catkin_ws/src
+WORKDIR /root/catkin_ws/src
+RUN git clone https://github.com/intel/ros2_object_msgs.git 
+RUN git clone -b ros2 https://github.com/intel/ros2_openvino_toolkit.git 
+RUN git clone -b ${VERSION} https://github.com/ros-perception/vision_opencv.git
+RUN git clone -b ros2-development https://github.com/IntelRealSense/realsense-ros.git 
+RUN apt-get install ros-${VERSION}-diagnostic-updater
+WORKDIR /root/catkin_ws
+RUN source /opt/ros/${VERSION}/setup.bash && source /opt/intel/openvino_2022/setupvars.sh && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release
```
Comment (huangjiafengx) [inline docker/Dockerfile:70]: No newline at end of file

File: docker/docker_instructions_ov2.0.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+Below steps have been tested on **Ubuntu 20.04**.
+
+## 1. Environment Setup
+* Install docker ([guide](https://docs.docker.com/engine/install/ubuntu/))
+
+## 2. Build docker image by dockerfile
+```
+cd ~/ros2_openvino_toolkit/docker/Dockerfile
+vi ~/ros2_openvino_toolkit/docker/Dockerfile
+docker build --build-arg ROS_VERSION=<EXPECT_ROS_BASE_IMAGE> --build-arg VERSION=<EXPECT_ROS_VERSION> --build-arg "HTTP_PROXY=set_your_proxy" -t ros2_openvino_202201 .
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply docker/docker_instructions_ov2.0.md:13]: Yes. It support dashing version. Humble are not supported due to OpenVINO and RealSense SDK are not supported on ubuntu22.04 right now. If you would like to test in humble, you'd better build and install OpenVINO and RealSense by source.

File: docker/Dockerfile (hunk 3 above)
Comment (ChloeWu1) [reply docker/Dockerfile:70]: Fixed. Thanks.

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# ignore the warning
+ARG DEBIAN_FRONTEND=noninteractive
+ARG APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=1
+RUN apt-get update && apt-get install --assume-yes apt-utils
+
+# install openvino 2022.1
+# https://docs.openvino.ai/2022.1/openvino_docs_install_guides_installing_openvino_apt.html
+RUN apt update && apt install --assume-yes curl gnupg2 lsb-release
+RUN curl -s https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB |apt-key add -
+RUN echo "deb https://apt.repos.intel.com/openvino/2022 bionic main" |tee /etc/apt/sources.list.d/intel-openvino-2022.list
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply docker/Dockerfile:26]: Fixed. Thanks.

File: docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+RUN source /opt/intel/openvino_2022/setupvars.sh 
+
+# install librealsense2
+RUN apt-get install -y --no-install-recommends \
+software-properties-common 
+# https://github.com/IntelRealSense/librealsense/blob/master/doc/distribution_linux.md
+# Make sure you set http-proxy in below commands if your environment needs.
+# RUN apt-key adv --keyserver-options http-proxy=your_proxy --keyserver keys.gnupg.net --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE || apt-key adv --keyserver-options http-proxy=your_proxy --keyserver hkp://keyserver.ubuntu.com:80 --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE 
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE || apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE
+RUN add-apt-repository "deb https://librealsense.intel.com/Debian/apt-repo bionic main" -u \
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply docker/Dockerfile:40]: Fixed. Thanks

File: docker/docker_instructions_ov2.0.md (hunk 2 above)
Comment (ChloeWu1) [reply docker/docker_instructions_ov2.0.md:73]: The download_model.lst and convert_model.lst have been added under the data folder. The docker instruction has been updated.

**Pull request merged**